### PR TITLE
fix(streams): add support `Float64Array` to `ReadableStreamByobReader`

### LINKED ISSUE #13176: ReadableStreamBYOBReader.read<Float64Array>(view: Float64Array) returns DataView
When I ran the following code, it resulted in unexpected behavior.

```typescript
Deno.test('1.0f from bytes.', () => {
  const b = new Uint8Array([
    0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0xF0, 0x3F,
  ]);
  const f = new Float64Array(b.buffer);
  a.assertEquals(String(f), "1"); // Okay
});

Deno.test('1.0f from bytes via BYOB stream.', async () => {
  const b = new Uint8Array([
    0x00, 0x00, 0x00, 0x00,
    0x00, 0x00, 0xF0, 0x3F,
  ]);
  const response = await fetch(URL.createObjectURL(new Blob([b])));
  if (!response.body) { throw new Error(); }

  const reader = response.body.getReader({ mode: 'byob' });
  const view = new Float64Array(1);
  // ReadableStreamBYOBReader.read<Float64Array>(view: Float64Array): Promise<ReadableStreamBYOBReadResult<Float64Array>>
  const result = await reader.read(view);
  a.assertEquals(result.done, false);
  a.assertEquals(Object.getPrototypeOf(result.value).constructor.name, 'Float64Array');
  // ...
  // failures:
  // 
  // 1.0f from bytes via BYOB stream.
  // AssertionError: Values are not equal:
  // 
  // 
  //    [Diff] Actual / Expected
  //
  //
  // -   DataView
  // +   Float64Array
  // ...
});
```

## Expect

ReadableStreamBYOBReader.read<Float64Array>(view: Float64Array) returns
`{ value: Float64Array, done: false }`

## Actual

ReadableStreamBYOBReader.read<Float64Array>(view: Float64Array) returns
`{ value: DataView, done: false }`

## Version

```bash
$ deno --version
deno 1.17.0 (release, x86_64-unknown-linux-gnu)
v8 9.7.106.15
typescript 4.5.2
```

### PR DESCRIPTION
**Pull request by petamoriken**
Closes #13176
This PR is an alternative of #17327. However, there is no update WPT test yet.

### CODE REVIEW

**Review by aapoalas — APPROVED** [review]
Looks good to me. Lack of a test would normally be a blocker for merging but I'll let the core team decide on this case since it's a matter of WPT test missing and not something Deno would normally implement directly.

**Review by crowlKats — COMMENTED** [review]
The alternative PR is blocked on WPT which i will work on soon, so this PR is blocked on WPT as well.

**Comment by bartlomieju** [comment]
Since this passes all WPT it appears to be non-breaking. Maybe we can land it without a test before #17327?

**Pull request merged**
